### ISSUE TITLE
Allow `.bmp` files to be exported without Alpha channel

### ISSUE THREAD

**Issue by Arecher** [aseprite/aseprite#3532]
### Aseprite and System version

* Aseprite 1.3-beta21-x64 (Steam)
* Windows 10

The latest changes to .bmp files and how they are exported seems to **always export with transparency enabled,** even if the sprite is fully solid.

For example, this Aseprite file will export to a .bmp that includes a transparent color in its palette.
![image](https://user-images.githubusercontent.com/10502115/191631556-1ae4fa59-103a-49f1-8aa4-9e8563205e32.png)

Seeing as it is known that not all applications support alpha channels in BMP files, I would like the option to export files into a .bmp format without transparency, as was the case before the latest bmp changes. Both for files that do not have any transparent pixels, and ones that do.

[bmpfiles.zip](https://github.com/aseprite/aseprite/files/9621136/bmpfiles.zip)

Here is the above mentioned Aseprite file, as well as two different .bmp files. One exported in 1.3.beta21, which shows a fully transparent color in the palette, when opened with Aseprite, and one exported in 1.2.18, which only shows black and white as colors. Both were exported from the same Aseprite file, which didn't have any transparent pixels.

**Comment by Gasparoken** [comment]
Thanks for this report. I'll fix it ASAP.

**Comment by Gasparoken** [comment]
> The latest changes to .bmp files and how they are exported seems to always export with transparency enabled, even if the sprite is fully solid.

A side note: if you have a single `Background Layer`, or if your `Color Mode` is `Indexed`, the BMP export results in the old behavior.

> Seeing as it is known that not all applications support alpha channels in BMP files, I would like the option to export files into a .bmp format without transparency.

Do you want this because you are having trouble opening Aseprite exported BMP files in other viewers/browsers?
Or is it just because you get the additional mask color in the palette?

We can add an option, but as a last resort.
To detect if the mask color is present before to start writing the BMP header we have to pre-scan the entire image, but this will add extra delay to the saving process. So, I'm not sure what is better:
- New option.
- Auto mask detection (will add delay to saving process)
- Do nothing, since there are a ways to get the old BMP saving behavior.

**Comment by Arecher** [comment]
> A side note: if you have a single Background Layer, or if your Color Mode is Indexed, the BMP export results in the old behavior.

Right, that's at least a way around it for the moment. That's nice! However, the .bmp files I export tend to be a generated image on a layer, within a main file, which has multiple layers and no indexed colors.

![image](https://user-images.githubusercontent.com/10502115/191810987-6e310ee2-1943-4522-a013-d65afd74c76d.png)


> Do you want this because you are having trouble opening Aseprite exported BMP files in other viewers/browsers?
Or is it just because you get the additional mask color in the palette?

I'm using Aseprite to create game assets, which I'm importing into my Godot 3.5 project. Godot has been flagging the new .bmp files as corrupt, which I've reported as a bug, as they should totally support these types of bmps too. However it will most likely be a long time before that gets fixed and merged, and it might not even happen in the current version. So I was hoping that I could use the old method of exporting through aseprite, and sidestep the limitation. (Ofcourse I can revert to an older Aseprite version, but I'd love to continue using the betas and getting access to the new features!)

I get that adding a new export option would be a huge step. And although scanning the image, and determining whether alpha channels are necesssary, would be what I would expect, I completely get that for 99% of the users this will never matter, so adding additional exporting time is most likely not worth it.
Perhaps adding a setting `disable Alpha Channels for bmp exports` in preferences,  might be a better option? That way users get to determine the export behaviour if they need to, while keeping the alpha channels as the default.

**Comment by Gasparoken** [comment]
If other popular apps have problems with BMP with alpha channel, I think we can add the option (as you said in the first place):
- "Allow alpha channel on BMP export" <-- unchecked as default.

This should appear as an option in the Warning window when you run "save as".

>  I completely get that for 99% of the users this will never matter, so adding additional exporting time is most likely not worth it.

Exactly what I thought 😄.

@dacap what do you think?

**Comment by dacap** [comment]
I think in the past we talked about this possible problem, because is the same problem Windows has with the clipboard. There is no easy way to handle the alpha channel with Windows bitmaps (and .bmp format). And each program does some heuristic (or nothing) to detect the alpha channel. I'd highly recommend to always use .png format when possible (but I'm not sure about Godot limitations).

Back to the problem, I think we should provide a checkbox to save or ignore the alpha channel, I think the option can be labeled as `[ ] Save alpha channel` (and it should only be visible for 32-bit bmps) in a new `BmpFormat::onAskUserForFormatOptions()` (and remember to provide a "show_alert" for "bmp" in pref.xml like in other formats so we can select an option for this checkbox and don't show the alert anymore).

**Comment by jordansavant** [comment]
Just opened Aseprite after a couple of months and my process broke. I am exporting bitmaps for data-centric portions of my sprite (eg hitboxes etc) and loading them with other scripts to build out data structures for my game. The BMP's now exported are considered invalid, such as with these GD functions https://www.php.net/manual/en/function.imagecreatefrombmp.php

I guess I will rewrite my process now :( but if your wondering if there is merit to this I think so.

**Comment by Gasparoken** [comment]
Thanks for this report @jordansavant. It is possibly related to issue https://github.com/aseprite/aseprite/issues/3534 (which is already solved but will be included in the next release).
Just to check, could you share any .bmp that had problems in your exporting process? (If you wish, you can send the file via support@aseprite.org)
